From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 11 Mar 2012 22:48:32 +0000
Subject: [PATCH] ia64: Define is_compat_task()

autofs now needs is_compat_task(), and we need to define it when
CONFIG_COMPAT is enabled.  Apparently hardware-assisted IA32 emulation
is broken and we should just disable it, but doing that now will break
ABI compatibility.
---
 arch/ia64/include/asm/compat.h |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/arch/ia64/include/asm/compat.h b/arch/ia64/include/asm/compat.h
index c8662cd..f5bafa6 100644
--- a/arch/ia64/include/asm/compat.h
+++ b/arch/ia64/include/asm/compat.h
@@ -4,6 +4,7 @@
  * Architecture specific compatibility types
  */
 #include <linux/types.h>
+#include <asm/system.h>
 
 #define COMPAT_USER_HZ	100
 
@@ -204,4 +205,11 @@ arch_compat_alloc_user_space (long len)
 	return (void __user *) (((regs->r12 & 0xffffffff) & -16) - len);
 }
 
+#ifdef CONFIG_COMPAT
+static inline int is_compat_task(void)
+{
+	return IS_IA32_PROCESS(task_pt_regs(current));
+}
+#endif
+
 #endif /* _ASM_IA64_COMPAT_H */
-- 
1.7.9.1

